internal/authz/authz_test.go

```diff
@@ -0,0 +1,315 @@
+package authz
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"github.com/openfga/openfga/internal/mocks"
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+func TestGetRelation(t *testing.T) {
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	mockServer := mocks.NewMockServerInterface(mockController)
+
+	authorizer := NewAuthorizer(&Config{StoreID: "test-store", ModelID: "test-model"}, mockServer, logger.NewNoopLogger())
+
+	t.Run("ReadAuthorizationModel", func(t *testing.T) {
+		result, err := authorizer.getRelation(ReadAuthorizationModel)
+		require.NoError(t, err)
+		require.Equal(t, CanCallReadAuthorizationModels, result)
+	})
+
+	t.Run("ReadAuthorizationModels", func(t *testing.T) {
+		result, err := authorizer.getRelation(ReadAuthorizationModels)
+		require.NoError(t, err)
+		require.Equal(t, CanCallReadAuthorizationModels, result)
+	})
```

Comment:
These tests are better written using [table-driven tests](https://go.dev/wiki/TableDrivenTests). We use this pattern a lot in this repo :)

```go
tests := map[string]struct {
		inputAPI       string
		outputRelation string
	}{
		`read_model`: {
			inputAPI:       ReadAuthorizationModel,
			outputRelation: CanCallReadAuthorizationModels,
		},
		`read-models`: {
			inputAPI:       ReadAuthorizationModels,
			outputRelation: CanCallReadAuthorizationModels,
		},
		// FILL ME
	}

	for testname, test := range tests {
		t.Run(testname, func(t *testing.T) {
			result, err := authorizer.getRelation(test.inputAPI)
			require.NoError(t, err)
			require.Equal(t, test.outputRelation, result)
		})
	}
```